pkg/internal/computeresources/tasklevel/tasklevel_test.go

```diff
@@ -0,0 +1,222 @@
+				Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+		}, {
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+		}},
+	}, {
+		desc: "both with requests and limits",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}, {
+				Name:    "2nd-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+		},
+		trs: v1beta1.TaskRunSpec{
+			ComputeResources: &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+				Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+			Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}, {
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+			Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}},
+	}, {
+		desc: "overwrite StepTemplate resources requirements",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}, {
+				Name:    "2nd-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+			StepTemplate: &v1beta1.StepTemplate{
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("500m"),
+						corev1.ResourceMemory: resource.MustParse("500Mi"),
+					},
+				},
+			},
+		},
+		trs: v1beta1.TaskRunSpec{
+			ComputeResources: &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("2"),
+					corev1.ResourceMemory: resource.MustParse("2Gi"),
+				},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+			},
+		}, {
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+			},
+		}},
+	}, {
+		desc: "with sidecar resource requirements",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+			Sidecars: []v1beta1.Sidecar{{
+				Name: "sidecar",
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU: resource.MustParse("750m"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU: resource.MustParse("1.5"),
+					},
+				},
+			}},
```

Comment:
Same comment as above; sidecars are ignored in `ApplyTaskLevelComputeResources` so they're not actually being tested here.

Reply:
ya. this testcase actually came from one of our previous threads
https://github.com/tektoncd/pipeline/pull/4877#discussion_r899154954

Reply:
closed - moved this testcase under pod_test.go